extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -7,251 +7,52 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.dropIndexesNotInSource.displayName
-			, this.deploymentOptions.ignoreFileAndLogFilePath.displayName
-			, this.deploymentOptions.ignoreExtendedProperties.displayName
-			, this.deploymentOptions.ignoreDmlTriggerState.displayName
-			, this.deploymentOptions.ignoreDmlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreDefaultSchema.displayName
-			, this.deploymentOptions.ignoreDdlTriggerState.displayName
-			, this.deploymentOptions.ignoreDdlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreCryptographicProviderFilePath.displayName
-			, this.deploymentOptions.verifyDeployment.displayName
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+		this.deploymentOptions = { ...defaultOptions };
 	}
 
 	/**
-	 * Gets the options checkbox check value
-	 * @returns string[][]
+	 * Initialize options data from deployment options for table component
+	 * Also preparing optionsValueNameLookup Map holding onchange checkbox values and property name
+	 * Returns data as [booleanValue, optionName]
 	 */
-	public getOptionsData(): string[][] {
-		let data: any = [];
-		this.optionsLookup = {};
-		this.optionsLabels.forEach(l => {
-			let checked: boolean = this.getDeployOptionUtil(l);
-			data.push([checked, l]);
-			this.optionsLookup[l] = checked;
+	public initializeOptionsData(): any[][] {
```

Comment:
Is there a reason this initialization of the lookup map is a public function and not something that's done in the constructor? As far as I can tell there will never be a time when we add/remove options after opening the dialog - only update their values. 

Then you could just have the getOptionsData function turn that into the array it needs for the table.

Reply:
They both needs to be initialize together, when a rest button clicked, we need to update the lookup map and get the latest data for the table. thats the reason, I have merged both and can also avoid multiple iterations and they both can be filled with single iteration on default options.

Reply:
Is it actually a valid scenario to have to potentially different options when you refresh? Isn't that something that should generally be static - unless there's a DacFX update or something that brings in additional options? Or at least not something I would expect to change during a users session. If that's the case not sure why you'd need to redo this logic each time since the existing values shouldn't be changing (just what their current assigned value is)

Reply:
New options only come with DacFx update! 

If the user modifies options, then the lookup table is to hold those updates until they click 'ok'.
If they reset table, needs to apply the latest data[][], if then click ok then sets the lookup data to deploymentOptions, else if cancel click dispose listener will dispose the change. In this reset+ok state either we can hold the lookup values until verify if the resetClicked is true and okClicked then set the deploymentOptions from lookup, (or) during reset only updating both data[][] and lookup! (this is what currently doing). Not sure if I'm on the same page that you are suggesting, not completely get your comment, Sorry!